packages/components/src/components/data-table/_data-table-core.scss

```diff
@@ -520,6 +520,12 @@
       height: auto;
     }
 
+    &:not(.bx--data-table--compact) td,
```

Comment:
Random, wasn't sure about this but would this be equivalent to:

```scss
&:not(.bx--data-table--compact, .bx--data-table--short, .bx--data-table--tall) td {
  // ...
}
```

?  Honestly wasn't sure how `not()` works lol

Reply:
Doesn't look like it's supported yet, but in the works 👍